src/lib.rs

```diff
@@ -447,11 +480,7 @@ impl Renderer {
             let buf_slice = bump_buf.slice(..);
             let (sender, receiver) = futures_intrusive::channel::shared::oneshot_channel();
             buf_slice.map_async(wgpu::MapMode::Read, move |v| sender.send(v).unwrap());
-            if let Some(recv_result) = receiver.receive().await {
-                recv_result?;
-            } else {
-                return Err("channel was closed".into());
-            }
+            receiver.receive().await.expect("channel was closed")?;
```

Comment:
So I was unsure if this could even happen and what it would mean for the end user if the channel was closed, so I replaced this with a panic

Reply:
Why are you unsure if it can happen? There should be a soundness "guarantee" with any panic. I'd rather not upgrade the error to a panic unless we are sure.

Reply:
I agree with @DasLixou that this error case is impossible, and so panicking is fine.

Anyway, it's in the legacy `_async` code, which we're no longer recommending, so I think this is fine